tests/test-gpg-signed-commit.sh

```diff
@@ -233,37 +233,52 @@ echo "ok verified with key2 primary expired"
 
 # Expire key2 subkey, wait until it's expired, export the public keys and
 # check again
-${GPG} --homedir=${tmpgpg_home} --quick-set-expire ${key2_fpr} seconds=1 ${key2_sub_fpr}
-sleep 2
-${GPG} --homedir=${tmpgpg_home} --export ${key1_id} ${key2_id} > ${tmpgpg_trusted_keyring}
-${OSTREE} show test2 > test2-show
-assert_file_has_content test2-show '^Found 2 signatures'
-assert_not_file_has_content test2-show 'public key not found'
-assert_file_has_content test2-show "${key1_id}"
-assert_file_has_content test2-show "${key2_sub_id}"
-assert_file_has_content test2-show 'Good signature from "Test Key 1 <>"'
-assert_file_has_content test2-show 'BAD signature from "Test Key 2 <>"'
-assert_file_has_content test2-show 'Key expired'
-assert_file_has_content test2-show "Primary key ID ${key2_id}"
-assert_file_has_content test2-show 'Primary key expired'
+if ! ${GPG} --homedir=${tmpgpg_home} --quick-set-expire ${key2_fpr} seconds=1 ${key2_sub_fpr}; then
```

Comment:
Hmm, could we do something like this instead here:

```
if ! (${GPG} --quick-set-expire || :) |& grep -q SUBKEY-FPRS; then
...
```

?

Reply:
Sure, I think that could be a little better. I usually shy away from scraping tool output since it will just break things when upstream changes them, but I guess that's better than papering over other issues. I did look at the usage output for my current gpg and the one on debian stretch and it is as you say.

gpg-2.2.12:
```
$ gpg --quick-set-expire 
usage: gpg [options] --quick-set-exipre FINGERPRINT EXPIRE [SUBKEY-FPRS]
```
gpg-2.1.18:
```
# gpg --quick-set-expire
usage: gpg [options] --quick-set-exipre FINGERPRINT EXPIRE
```

Reply:
For some reason `(${GPG} --quick-set-expire || :) |&` didn't actually send the output to `grep` when I ran it through the test suite (it did work in my interactive shell). Instead I added an explicit `2>&1` in the `${GPG}` command.